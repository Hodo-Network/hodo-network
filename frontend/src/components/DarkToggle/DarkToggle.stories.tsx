import { Story, Meta } from "@storybook/react";

import { DarkToggle } from "./DarkToggle";

export default {
  title: "Components/DarkToggle",
  component: DarkToggle,
} as Meta;

const Template: Story = (args) => <DarkToggle {...args} />;

export const Default = Template.bind({});
Default.args = {};
