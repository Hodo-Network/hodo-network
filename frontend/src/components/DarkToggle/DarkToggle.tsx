import { MoonIcon, SunIcon } from "@heroicons/react/solid";
import { useDark } from "../../hooks";

export const DarkToggle = () => {
  const [dark, toggleDark] = useDark();

  return (
    <button
      type='button'
      onClick={toggleDark}
      className='inline-flex items-center p-2 rounded-full'>
      {dark ? (
        <SunIcon className='h-6 w-6' />
      ) : (
        <MoonIcon className='h-6 w-6' />
      )}
    </button>
  );
};
