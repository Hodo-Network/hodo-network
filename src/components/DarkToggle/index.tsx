import { MoonIcon, SunIcon } from "@heroicons/react/solid";
import { useDark } from "../../hooks";

export default function DarkToggle() {
  const [dark, toggleDark] = useDark();

  return (
    <button
      type='button'
      onClick={toggleDark}
      className='inline-flex items-center p-1 rounded-full'>
      {dark ? (
        <SunIcon className='h-6 w-6 text-white' />
      ) : (
        <MoonIcon className='h-6 w-6 text-gray-900' />
      )}
    </button>
  );
}
