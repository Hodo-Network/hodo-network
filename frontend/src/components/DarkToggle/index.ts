import { DarkToggle } from './DarkToggle';
export { DarkToggle };
