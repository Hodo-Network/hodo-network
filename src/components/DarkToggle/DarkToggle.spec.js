import ReactDOM from "react-dom";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import DarkToggle from "./index";

afterEach(cleanup);

test("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<DarkToggle />, div);
});

test("matches snapshot", () => {
  const tree = renderer.create(<DarkToggle />).toJSON();
  expect(tree).toMatchSnapshot();
});
