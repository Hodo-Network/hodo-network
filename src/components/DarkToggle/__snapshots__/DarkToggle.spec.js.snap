// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="inline-flex items-center p-1 rounded-full"
  onClick={[Function]}
  type="button"
>
  <svg
    className="h-6 w-6 text-gray-900"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M17.293 13.293A8 8 0 016.707 2.707a8.001 8.001 0 1010.586 10.586z"
    />
  </svg>
</button>
`;
